CesiumGltf/include/CesiumGltf/MetadataConversions.h

```diff
@@ -152,56 +174,59 @@ struct MetadataConversions<
 };
 
 /**
- * @brief Converts from std::string_view to a signed integer.
+ * @brief Converts from std::string to a signed integer.
  */
 template <typename TTo>
 struct MetadataConversions<
     TTo,
-    std::string_view,
+    std::string,
     std::enable_if_t<
         CesiumGltf::IsMetadataInteger<TTo>::value && std::is_signed_v<TTo>>> {
   /**
-   * @brief Converts the contents of a std::string_view to a signed integer.
-   * This assumes that the entire std::string_view represents the number, not
+   * @brief Converts the contents of a std::string to a signed integer.
+   * This assumes that the entire std::string represents the number, not
    * just a part of it.
    *
    * This returns std::nullopt if no number is parsed from the string.
    *
-   * @param from The std::string_view to parse from.
+   * @param from The std::string to parse from.
    */
-  static std::optional<TTo> convert(const std::string_view& from) {
+  static std::optional<TTo> convert(const std::string& from) {
     if (from.size() == 0) {
       // Return early. Otherwise, empty strings will be parsed as 0, which is
       // misleading.
       return std::nullopt;
     }
 
-    // Amazingly, C++ has no* string parsing functions that work with strings
-    // that might not be null-terminated. So we have to copy to a std::string
-    // (which _is_ guaranteed to be null terminated) before parsing.
-    // * except std::from_chars, but compiler/library support for the
-    //   floating-point version of that method is spotty at best.
-    std::string temp(from);
+    errno = 0;
 
     char* pLastUsed;
-    int64_t parsedValue = std::strtoll(temp.c_str(), &pLastUsed, 10);
-    if (pLastUsed == temp.c_str() + temp.size()) {
-      // Successfully parsed the entire string as an integer of this type.
-      return CesiumUtility::losslessNarrow<TTo, int64_t>(parsedValue);
+    int64_t parsedValue = std::strtoll(from.c_str(), &pLastUsed, 10);
```

Comment:
From browsing the documentation on `strtoll`, I do see `ERANGE` as one of the error values. I don't see `EINVAL` mentioned.

Also, I see that if no conversion can be performed, `0` or ` LONG_MAX, LONG_MIN, LLONG_MAX or LLONG_MIN` could be returned too.